python/dgl/sparse/sparse_matrix.py

```diff
@@ -479,6 +479,128 @@ def is_diag(self):
         """Returns whether the sparse matrix is a diagonal matrix."""
         return self.c_sparse_matrix.is_diag()
 
+    def rowwise_select(self, ids: torch.Tensor):
+        """Returns a row-wise sparse matrix selected by ids.
+
+        Parameters
+        ----------
+        ids : tensor.Tensor
+            The indices indicate the rows selected from matrix.
+
+        The function has not support autograd.
+
+        Returns
+        -------
+        SparseMatrix
+            The row-wise sparse matrix only contains the rows selected.
+
+        Examples
+        --------
+        >>> indices = torch.tensor([0, 1, 1, 2, 3, 4], [0, 2, 4, 3, 5, 0]])
+        >>> val = torch.tensor([0, 1, 2, 3, 4, 5])
+        >>> A = dglsp.spmatrix(indices, val)
+        >>> row_ids = torch.tensor([0, 1, 4])
+        >>> B = A.rowwise_select(row_ids)
+        >>> B
+        SparseMatrix(indices=tensor([[0, 1, 1, 2],
```

Comment:
The example looks incorrect. Please rerun the example after your implement this interface.

Reply:
This is my example:
| | 0 | 1 | 2 | 3 | 4 | 5 |
| ---- | ---- | :---- | ----: | :----: | ---- | ---- |
0 | 0 |   |    |    |     |    
1 |    |    | 1 |    |  2  |    
2 |   |    |    | 3 |     |    
3 |   |    |    |    |     |  4
4 | 5 |    |    |    |     |    

if we select [0,1,4] rows:
| | 0 | 1 | 2 | 3 | 4 | 5 |
| ---- | ---- | :---- | ----: | :----: | ---- | ---- |
0 | 0 |   |    |    |     |    
1 |    |    | 1 |    |  2  |    
4 | 5 |    |    |    |     |    

so answer is: 
SparseMatrix(indices=tensor([[0, 1, 1, 2],[0, 2, 4, 0]]),
                     values=tensor([0, 1, 2, 5]),
                     shape=(3, 6), nnz=4)

Reply:
Got you. The row ids are actually reordered.